src/dapps/BuySubdomain/BuySubdomain.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+  <div class="buy-subdomain-container">
+    <back-button/>
+    <div class="buy-subdomain-content">
+      <div class="buy-subdomain-form-container">
+        <p>Sub Domain</p>
+        <div class="form">
+          <div class="subdomain-input">
+            <input
+              type="text"
+              placeholder="Please Enter Sub Domain Name"
+              @input="debounceInput"
+            >
+            <button
+              type="button"
+              @click="query">Check</button>
+          </div>
+        </div>
+        <div
+          v-show="results.length > 0"
+          class="result-section">
+          <p>All Sub domains</p>
+          <div class="results-container">
+            <div
+              v-for="item in sortedResults"
+              :key="domainName+item.domain"
+              :class="[ item.active? '':'disabled','result-item']">
+              <span>{{ domainName }}.{{ item.domain }}.eth</span>
+              <span>
+                <span class="amt">{{ $store.state.web3.utils.fromWei(item.price, 'ether') }} </span>
+                <span class="currency">ETH </span>
+                <button @click="buyDomain(item)" >
+                  <span v-if="item.active">
+                    Buy
+                  </span>
+                  <span v-else>
+                    <i class="fa fa-times" />
+                  </span>
+                </button>
+              </span>
+            </div>
+          </div>
+        </div>
+      </div>
+      <div>
+        <interface-bottom-text
+          :link-text="$t('interface.learnMore')"
+          :question="$t('interface.haveIssues')"
+          link="/"/>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import InterfaceBottomText from '@/components/InterfaceBottomText';
+import BackButton from '@/layouts/InterfaceLayout/components/BackButton';
+import SubdomainAbi from '@/helpers/subdomainAbi.js';
+import domains from './domains.json';
+import normalise from '@/helpers/normalise';
+import BN from 'bignumber.js';
+import web3 from 'web3';
+
+export default {
+  components: {
+    'interface-bottom-text': InterfaceBottomText,
+    'back-button': BackButton
+  },
+  data() {
+    return {
+      subdomainContract: function() {},
+      ensContract: function() {},
+      results: [],
+      domainName: '',
+      knownRegistrarInstances: {},
+      referrerAddress: '0xDECAF9CD2367cdbb726E904cD6397eDFcAe6068D'
```

Comment:
may be we should define our donation address globally that way we can easily access it everywhere

Reply:
I like that. I'll fix it.